File: ts-sdk/src/sdk/models/shared/workflow.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { WorkflowConfig } from "./workflowconfig";


export class Workflow extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=config" })
  config: WorkflowConfig;

  @SpeakeasyMetadata({ data: "json, name=createdAt" })
  createdAt: Date;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id: string;

  @SpeakeasyMetadata({ data: "json, name=updatedAt" })
  updatedAt: Date;
}

File: python-sdk/src/formanceapi/models/shared/workflow.py
import dataclasses
import dateutil.parser
from ..shared import workflowconfig as shared_workflowconfig
from dataclasses_json import dataclass_json
from datetime import datetime
from formanceapi import utils
from marshmallow import fields


@dataclass_json
@dataclasses.dataclass
class Workflow:
    config: shared_workflowconfig.WorkflowConfig = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('config') }})
    created_at: datetime = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('createdAt'), 'encoder': utils.datetimeisoformat(False), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    updated_at: datetime = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('updatedAt'), 'encoder': utils.datetimeisoformat(False), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    